photon-lib/src/main/scala/com/linkedin/photon/ml/hyperparameter/estimators/kernels/StationaryKernel.scala

```diff
@@ -71,24 +86,64 @@ abstract class StationaryKernel(
     val ls = expandDimensions(lengthScale, x1.cols)
     val dists = pairwiseDistances(x1(*,::) / ls, x2(*,::) / ls)
 
-    fromPairwiseDistances(dists)
+    amplitude * fromPairwiseDistances(dists)
   }
 
   /**
    * Returns the kernel parameters as a vector
    *
    * @return the kernel parameters
    */
-  override def getParams: DenseVector[Double] = log(lengthScale)
+  override def getParams: DenseVector[Double] = DenseVector.vertcat(DenseVector(amplitude, noise), lengthScale)
 
   /**
-   * Returns the kernel parameter bounds
+   * Computes the log likelihood of the kernel parameters
    *
-   * @return the kernel parameter bounds
+   * @param x the observed features
+   * @param y the observed labels
+   * @return the log likelihood
    */
-  override def getParamBounds: (Double, Double) = {
-    val (upperBound, lowerBound) = lengthScaleBounds
-    (log(upperBound), log(lowerBound))
+  override def logLikelihood(x: DenseMatrix[Double], y: DenseVector[Double]): Double = {
+    // Bounds checks
+    if (amplitude < 0.0 ||
+        noise < 0.0 ||
+        any(lengthScale :< 0.0)) {
+      return Double.NegativeInfinity
+    }
+
+    // Tophat prior for length scale
+    if (any(lengthScale :> lengthScaleMax)) {
```

Comment:
This makes it very strict. If any of the l > 2, we are returning -\infty. Can this be done in a more adaptive way? The variance in the scale is usually unknown.

Reply:
This is also defaulted to spearmint's implementation. Maybe we could add a lognormal prior term for length scale? That would be less strict, but would still require a pre-set scaling factor.

Reply:
Let's keep a note of these things and come back to these in later iterations.